Appdata related patches

sabriunal opened the pull request:
### appdata: Use appstreamcli for appdata validation 

appstream-util is obsoleted by appstreamcli.

### appdata: Add categories

These tags may improve Software Center accessibility.

More information: https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-categories

yakushabb committed:
appdata: Use appstreamcli for appdata validation

appstream-util is obsoleted by appstreamcli.

ghost commented:
Michael Catanzaro recommended the use of appstreamcli in [this comment](https://gitlab.gnome.org/World/eartag/-/merge_requests/95#note_1874726).

yakushabb committed:
appdata: Add categories

These tags may improve Software Center accessibility.

More information: https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-categories

ghost commented:
Additionally, the appdata and desktop files don't contain any keywords. We may consider adding some.

- https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-keywords
- https://gitlab.gnome.org/GNOME/nautilus/-/blob/main/data/org.gnome.Nautilus.desktop.in.in?ref_type=heads#L5

ADBeveridge commented:
LGTM! Thanks again Sabri.
